javaslang/src/main/java/javaslang/Lazy.java

```diff
@@ -135,14 +133,11 @@ private Lazy(Supplier<? extends T> supplier) {
      */
     @Override
     public T get() {
-        // using a local var speeds up the double-check idiom by 25%, see Effective Java, Item 71
-        Supplier<? extends T> tmp = supplier;
-        if (tmp != null) {
+        if ((value == null) && (supplier != null)) {
```

Comment:
Hi @jbgi, inlining by keeping it small is a great idea!!
I also think that the read of `value` is dirty.
The local var `T v` most probably isn't needed because `value` isn't volatile and reads are cheap.
I will further play around with different solutions....